.github/workflows/actions.yml

```diff
@@ -0,0 +1,84 @@
+name: CI
+
+on:
+  push:
+    branches: '**'
+    tags: 'v*'
+  pull_request:
+    branches: '**'
+
+env:
+  BUILD_VERSION: 23.6.0.${{ github.run_number }}
```

Comment:
Could we maybe use the github.ref_name?  See https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables
`BUILD_VERSION: ${{ github.ref_name }}.${{ github.run_number }}`

Reply:
Thanks for adding a comment here, I realized that my fishy fix in https://github.com/cqse/teamscale-profiler-dotnet/pull/179/commits/44ba4a34a85674d3522e3d5f848938e7d73de75c on the `resource.rc` is incorrect 🙈 
The problem is that we should ensure that the BUILD_VERSION is in the format of `{2-byte-number}.{2-byte-number}.{2-byte-number}.{2-byte-number}` which might already not be the case (as it was breaking the builds after the "Patch C++ Version Info" step). I will take a look.

Reply:
It turned out that the Powershell expression was a bit off and set the version to empty string, now it should be fixed. :grin:

I applied the change on calculating the `BUILD_VERSION`, and added a check so the build fails early if the version number is off. It also cuts off the `v` prefix of the tag (there is no built-in GitHub Action expression for it, see https://github.com/orgs/community/discussions/26625)